### PR TITLE
[BL-168] Fixed sync for etcd v3

### PR DESCRIPTION
**Pull request by michalzurawski**
Sync fetch for etcdv3 behaves differently than in version 2 when multiple keys could be obtained. Previously only one key was returned

### CODE REVIEW

**Comment by michalzurawski** [comment]
Please do not merge this PR, some integration tests have to be performed before

File: sync/etcdv3/etcd_test.go [hunk 1]
```diff
@@ -285,6 +291,98 @@ func TestWatchWithRevision(t *testing.T) {
...
+
+func checkNode(node *gohan_sync.Node, key, value string, children int, t *testing.T) {
+	if node.Key != key {
+		t.Fatalf("expected key %s has %s", key, node.Key)
+	}
+	if node.Value != value {
+		t.Fatalf("expected value %s has %s", value, node.Value)
+	}
+	if len(node.Children) != children {
+		t.Fatalf("expected to has %d children has %s", children, len(node.Children))
```
Comment (yudai) [inline sync/etcdv3/etcd_test.go]: `%d`?

File: sync/etcdv3/etcd_test.go [hunk 2]
```diff
@@ -71,51 +73,55 @@ func TestEmptyUpdate(t *testing.T) {
...
-	node, err := sync.Fetch(base)
-	if err != nil {
-		t.Fatalf("unexpected error")
-	}
+	mustUpdate("/a/b/c/d/1", "test1")
+	mustUpdate("/a/b/c/d/2", "test2")
+	mustUpdate("/a/b/c/d/3", "test3")
+	mustUpdate("/a/b/e/d/1", "test4")
+	mustUpdate("/a/b/e/d/2", "test5")
+	mustUpdate("/a/b/e/d/3", "test6")
```
[the remark this answers is not in the document]
Comment (yudai) [reply sync/etcdv3/etcd_test.go]: I tested locally and confirmed `/abcd` breaks the test

File: sync/etcdv3/etcd_test.go [hunk 3]
```diff
@@ -285,6 +291,98 @@ func TestWatchWithRevision(t *testing.T) {
...
+		t.Fatalf("unexpected error")
+	}
+	err = sync.Update("/path/to/elsewhere", "test2")
+	if err != nil {
+		t.Fatalf("unexpected error")
+	}
+	err = sync.Update("/path/notto/elsewhere", "test3")
+	if err != nil {
+		t.Fatalf("unexpected error")
+	}
```
Comment (yudai) [inline sync/etcdv3/etcd_test.go]: maybe better to add `/pathnottobeincluded` as well to confirm it's not included in the output.

**Comment by michalzurawski** [comment]
@yudai thanks a lot for comments, fixed that and added separate test for that

**Pull request merged**